### PR TITLE
chore(DTFS2-7320): dry cypress commands - assertText

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR introduces a new `assertText` cypress command.

Now, for example, instead of having this everywhere:

```js
applicationDetails
  .mainHeading()
  .invoke('text')
  .then((text) => {
    expect(text.trim()).to.equal('Application Details');
  });
```

We can simply do the following:

```js
cy.assertText(applicationDetails.mainHeading(), 'Application Details');
```

## Resolution :heavy_check_mark:
- Add `assertText` cypress command to all E2E suites.
- Update all E2E instances of tests with `invoke('text')`, `text.trim()` and `equal` / `.to.eq` to use the new cypress command.

### CODE REVIEW

File: e2e-tests/gef/cypress/e2e/application/application-details.spec.js [hunk 1]
```diff
@@ -62,12 +62,8 @@ context('Application Details Page', () => {
...
       applicationDetails.applicationDetailsPage();
       applicationDetails.captionHeading();
-      applicationDetails
-        .mainHeading()
-        .invoke('text')
-        .then((text) => {
-          expect(text.trim()).to.equal('Application Details');
-        });
+
+      cy.assertText(applicationDetails.mainHeading(), 'Application Details');
```
Comment (Zainzzkk) [inline e2e-tests/gef/cypress/e2e/application/application-details.spec.js:66]: can we add a tech debt ticket for content-strings?

File: e2e-tests/gef/cypress/e2e/application/application-details.spec.js [hunk 2]
```diff
@@ -168,12 +164,7 @@ context('Application Details Page', () => {
     });
 
     it('displays the correct submission type heading', () => {
-      applicationDetails
-        .mainHeading()
-        .invoke('text')
-        .then((text) => {
-          expect(text.trim()).to.equal('Application Details');
-        });
+      cy.assertText(applicationDetails.mainHeading(), 'Application Details');
```
Comment (Zainzzkk) [inline e2e-tests/gef/cypress/e2e/application/application-details.spec.js:167]: repeated twice, const for Application Details?

File: e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js [hunk 3]
```diff
@@ -68,19 +68,10 @@ context('A maker issues facilities, submits to checker; checker submits deal to
...
-        expect(text.trim()).to.equal('Completed');
-      });
 
-    loanRow
-      .loanStatus()
-      .invoke('text')
-      .then((text) => {
-        expect(text.trim()).to.equal('Completed');
-      });
+    cy.assertText(bondRow.bondStatus(), 'Completed');
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js:72]: is there a const for these?

File: e2e-tests/portal/cypress/e2e/journeys/checker/return-a-deal-to-maker/return-deal-to-maker.spec.js [hunk 4]
```diff
@@ -62,43 +62,21 @@ context('A checker selects to return a deal to maker from the view-contract page
...
-    contract
-      .previousStatus()
-      .invoke('text')
-      .then((text) => {
-        expect(text.trim()).to.equal("Ready for Checker's approval");
-      });
+
+    cy.assertText(contract.status(), "Further Maker's input required");
+
+    cy.assertText(contract.previousStatus(), "Ready for Checker's approval");
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/checker/return-a-deal-to-maker/return-deal-to-maker.spec.js:73]: status constants?

File: e2e-tests/tfm/cypress/e2e/journeys/case-underwriting/underwriter-managers-decision-decline.spec.js [hunk 5]
```diff
@@ -51,44 +51,17 @@ context('Case Underwriting - Pricing and risk', () => {
...
-    pages.managersDecisionPage
-      .decisionDateTime()
-      .invoke('text')
-      .then((text) => {
-        const todayFormatted = new Date().toLocaleString('en-GB', { year: 'numeric', month: 'long', day: '2-digit' });
+    cy.assertText(pages.managersDecisionPage.decisionMadeBy(), expectedName);
 
-        expect(text.trim()).contains(todayFormatted);
-      });
+    cy.assertText(pages.managersDecisionPage.decisionDateTime(), new Date().toLocaleString('en-GB', { year: 'numeric', month: 'long', day: '2-digit' }));
```
Comment (Zainzzkk) [inline e2e-tests/tfm/cypress/e2e/journeys/case-underwriting/underwriter-managers-decision-decline.spec.js:61]: could this be a const? more readable (the date)

File: e2e-tests/tfm/cypress/e2e/journeys/case-underwriting/underwriter-managers-decision.spec.js [hunk 6]
```diff
@@ -218,54 +218,21 @@ context("Case Underwriting - Underwriter Manager's decision - Submit Form", () =
...
-    pages.managersDecisionPage
-      .decisionDateTime()
-      .invoke('text')
-      .then((text) => {
-        const todayFormatted = new Date().toLocaleString('en-GB', { year: 'numeric', month: 'long', day: '2-digit' });
+    cy.assertText(pages.managersDecisionPage.decisionMadeBy(), expectedName);
 
-        expect(text.trim()).contains(todayFormatted);
-      });
+    cy.assertText(pages.managersDecisionPage.decisionDateTime(), new Date().toLocaleString('en-GB', { year: 'numeric', month: 'long', day: '2-digit' }));
```
Comment (Zainzzkk) [inline e2e-tests/tfm/cypress/e2e/journeys/case-underwriting/underwriter-managers-decision.spec.js:229]: same as above

File: e2e-tests/tfm/cypress/support/commands.js [hunk 7]
```diff
@@ -1,6 +1,9 @@
 // Preserve session cookie
 Cypress.Commands.add('saveSession', require('./utils/saveSession'));
 
+// Assert an element has some exact text
```
Comment (Zainzzkk) [inline e2e-tests/tfm/cypress/support/commands.js:4]: do we need this comment here?

File: e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js (hunk 3 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js:72]: I'm not sure, but status constants (and/or usage of them) should be handled separately. This PR is purely E2E, introducing an "assert text" cypress command

File: e2e-tests/portal/cypress/e2e/journeys/checker/return-a-deal-to-maker/return-deal-to-maker.spec.js (hunk 4 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/e2e/journeys/checker/return-a-deal-to-maker/return-deal-to-maker.spec.js:73]: Same as above

File: e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js (hunk 3 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/journeys/checker/issued-specs-new/submit-MIN-deal-with-issued-facility/submit-MIN-deal-with-issued-facilities.spec.js:72]: tech debt ticket for all status strings to come from constants please

File: e2e-tests/gef/cypress/e2e/application/application-details.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/gef/cypress/e2e/application/application-details.spec.js:66]: Created this ticket => https://ukef-dtfs.atlassian.net/browse/DTFS2-7338

File: e2e-tests/gef/cypress/e2e/application/application-details.spec.js (hunk 2 above)
Comment (ttbarnes) [reply e2e-tests/gef/cypress/e2e/application/application-details.spec.js:167]: I think this should be part of a wider ticket for content strings or similar.

Would not want to have a different "main heading content string/const" pattern in place specifically for this page. Also, this PR is purely about replacing cypress "text/trim/to.equal" text assertions with a new DRY command.

File: e2e-tests/tfm/cypress/support/commands.js (hunk 7 above)
Comment (ttbarnes) [reply e2e-tests/tfm/cypress/support/commands.js:4]: I'm not sure what is best here. Need to declare this command 3x times for different microservices - some command files have comments above each command, others don't.

I have added comment for each command as this seems to be the more prominent approach.

I do think that no command should require a description, it should be self describing.

What do you think (for this PR/command file)?

Comment (Zainzzkk) [reply e2e-tests/tfm/cypress/support/commands.js:4]: As its already a pattern here, leave it.  But I think the actual function should have a description and not the commands file

**Review by Zainzzkk — DISMISSED** [review]

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**